app/controllers/application_controller.rb

```diff
@@ -36,7 +36,7 @@ def patron_info_response
 
     def authenticate_user!
       redirect_to root_url unless current_user?
-      renew_session_token if patron.stale?
+      renew_session_token if current_user? && patron.stale?
```

Comment:
🤔 so, curious why you added a check for `current_user?`? Doesn't the `unless` above mean that `current_user?` resolves to `true`? Is this redundant? 

Also, your commit says "

"since redirect_to does not return, this line will be executed so check if current_user, otherwise errors out with undefined patron." Here is my understanding: There are 3 scenarios:

1) `current_user?` is false - redirect this to the root_url (which will start up authentication)
2) `current_user?` is true (which is implied in the world that exists after the first line of the method):

In 2) either the patron.stale? is true or false. If it is true, then the renew_session_token method is called. If it's false, then the method has basically done nothing and returns `nil`.

Is that wrong? I'm not sure I see where erroring out with an undefined patron happens?

Reply:
we chat about this and decided that we will use `return redirect_to` since we both like that way better.

Reply:
Just for future me if I'm re-reading, without `return` the second line in the method gets processed. Thanks @banukutlu